src/ssl.c

```diff
@@ -41272,9 +41272,7 @@ int wolfSSL_get_state(const WOLFSSL* ssl)
 }
 #endif /* HAVE_LIGHTY || HAVE_STUNNEL || WOLFSSL_MYSQL_COMPATIBLE */
 
-#if defined(OPENSSL_ALL) || defined(WOLFSSL_ASIO) || defined(WOLFSSL_HAPROXY) \
-    || defined(WOLFSSL_NGINX) || defined(WOLFSSL_QT)
-
+#if defined(OPENSSL_ALL) || defined(OPENSSL_EXTRA) || defined(OPENSSL_EXTRA_X509_SMALL)
```

Comment:
We have customers that strictly rely on the footprint on OPENSSL_EXTRA_X509_SMALL not increasing between releases. What is the purpose for introducing new code sections to the X509_SMALL build?

Reply:
The reason is described in the commit message for this commit:
"match preproc defs around wolfSSL_ctrl() in wolfssl/ssl.h"

Please indicate whether the preprocessor wrapping in the `wolfssl/ssl.h` is correct, or if the preprocessor wrapping in `src/ssl.c` is correct.

Reply:
Glenn,

Let's align on the ssl.c (prototype in the header doesn't increase footprint for X509_SMALL but including the function source does)

Proposed alternative:

```
diff --git a/wolfssl/ssl.h b/wolfssl/ssl.h
index b2e8a9836..5285d4456 100644
--- a/wolfssl/ssl.h
+++ b/wolfssl/ssl.h
@@ -3310,7 +3310,6 @@ WOLFSSL_LOCAL int NIDToEccEnum(int n);
 /* end of object functions */
 
 WOLFSSL_API unsigned long wolfSSL_ERR_peek_last_error_line(const char **file, int *line);
-WOLFSSL_API long wolfSSL_ctrl(WOLFSSL* ssl, int cmd, long opt, void* pt);
 WOLFSSL_API long wolfSSL_CTX_ctrl(WOLFSSL_CTX* ctx, int cmd, long opt,void* pt);
 WOLFSSL_API long wolfSSL_CTX_callback_ctrl(WOLFSSL_CTX* ctx, int cmd, void (*fp)(void));
 WOLFSSL_API long wolfSSL_CTX_clear_extra_chain_certs(WOLFSSL_CTX* ctx);
@@ -3340,6 +3339,11 @@ WOLFSSL_API int wolfSSL_check_private_key(const WOLFSSL* ssl);
 #endif /* !NO_CERTS */
 #endif /* OPENSSL_ALL || OPENSSL_EXTRA || OPENSSL_EXTRA_X509_SMALL */
 
+#if defined(OPENSSL_ALL) || defined(WOLFSSL_ASIO) || defined(WOLFSSL_HAPROXY) \
+    || defined(WOLFSSL_NGINX) || defined(WOLFSSL_QT)
+WOLFSSL_API long wolfSSL_ctrl(WOLFSSL* ssl, int cmd, long opt, void* pt);
+#endif
+
 #ifdef WOLFSSL_WPAS_SMALL
     /* WPA Supplicant requires GEN_ values */
     #include <wolfssl/openssl/x509v3.h>
```

Reply:
implemented requested change